services/bit/bit-components.service.js

```diff
@@ -0,0 +1,60 @@
+'use strict'
+
+const Joi = require('joi')
+const { BaseJsonService } = require('..')
+const { metric } = require('../text-formatters')
+const { nonNegativeInteger } = require('../validators')
+const { downloadCount } = require('../color-formatters')
+
+const collectionSchema = Joi.object({
+  payload: Joi.object({
+    totalComponents: nonNegativeInteger,
+  }).required(),
+}).required()
+
+module.exports = class bitComponents extends BaseJsonService {
+  async fetch({ owner, scope }) {
+    const url = `https://api.bit.dev/scope/${owner}/${scope}/`
+    return this._requestJson({
+      url,
+      schema: collectionSchema,
+      errorMessages: {
+        404: 'collection not found',
+      },
+    })
+  }
+
+  static render({ count }) {
+    return { message: metric(count), color: downloadCount(count) }
+  }
+
+  async handle({ owner, scope }) {
+    const json = await this.fetch({ owner, scope })
+    return this.constructor.render({ count: json.payload.totalComponents })
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'components' }
+  }
+
+  static get category() {
+    return 'other'
+  }
+
+  static get route() {
+    return {
+      base: 'bit/collection/total-components',
+      pattern: ':owner/:scope',
```

Comment:
One final question, is  `scope` the same as the collection name and is that something that bit.dev users will know? If so then this is good, but if not, would `:collection` be more clear?

Reply:
scope and collection is the same, but the right name for this now is collection, 
I changed it to `collection`,
 thanks :)